src/app/app.component.ts

```diff
@@ -1,10 +1,99 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { FruitService } from './fruit.service';
+import { Item } from './models/item';
+import { CartService } from './cart.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css']
+  styleUrls: ['./app.component.css'],
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
   title = 'angular-green-grocers';
+  isFruitButtonClicked = false;
+  isVegetableButtonClicked = false;
+  isEverythingButtonClicked = true;
+  isSortedByPrice = false;
+  isSortedByName = false;
+  priceSorting = 'Price ▲';
+  nameSorting = 'Name ▲';
+
+  vegetables: Item[] = [];
+  fruits: Item[] = [];
+  filteredGroceries: Item[] = [];
+  allGroceries: Item[] = [];
+  cart = new Map();
+
+  constructor(
+    private readonly fruitService: FruitService,
+    private readonly cartService: CartService
+  ) {
+    this.cartService.getAllItems().subscribe((cart) => {
```

Comment:
Remember we need to store the subscriptions in order to unsubscribe to prevent memory leaks